### PR TITLE
feat(dbal): insert ignore conflict method for MySQL and SQLite

### PR DESCRIPTION
**Pull request by Altahrim**
## Summary

Add ignore conflicts method for MySQL
Similar to PgSQL: https://github.com/nextcloud/server/blob/master/lib/private/DB/AdapterPgSql.php#L26-L37

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by artonge — APPROVED** [review]

**Review by come-nc — COMMENTED** [review]
On conflict this ones updates the row while pgsql implementation will not touch it, no?

**Comment by Altahrim** [comment]
> On conflict this ones updates the row while pgsql implementation will not touch it, no?

Yep, see above comment :)

**Comment by solracsf** [comment]
> On conflict this ones updates the row while pgsql implementation will not touch it, no?

Can this be an option?
https://mariadb.com/kb/en/insert-ignore/

**Comment by Altahrim** [comment]
> Can this be an option?
> [mariadb.com/kb/en/insert-ignore](https://mariadb.com/kb/en/insert-ignore/)

For me it ignores too many errors. I would prefer to avoid this.

File: lib/private/DB/AdapterMySQL.php [hunk 1]
```diff
@@ -35,4 +35,20 @@ protected function getCollation(): string {
...
+	public function insertIgnoreConflict(string $table, array $values): int {
+		$builder = $this->conn->getQueryBuilder();
+		$builder->insert($table);
+		$updates = [];
+		foreach ($values as $key => $value) {
+			$builder->setValue($key, $builder->createNamedParameter($value));
+		}
+
+		$field = key($values);
+		return $this->conn->executeStatement(
```
Comment (Altahrim) [inline lib/private/DB/AdapterMySQL.php:48]: This method doesn't return the `affected_rows` as MySQL do in CLI
That's why the tests are failing. If anyone have an idea, I am interested :) 

Example from the tests:
```sql
INSERT INTO oc_twofactor_providers (provider_id, uid, enabled) VALUES ('test', 'user123', 0) ON DUPLICATE KEY UPDATE provider_id = provider_id;
-- MySQL CLI: Query OK, 1 row affected (0.018 sec)
-- PHP returns: 1
INSERT INTO oc_twofactor_providers (provider_id, uid, enabled) VALUES ('test', 'user123', 1) ON DUPLICATE KEY UPDATE provider_id = provider_id;
-- MySQL CLI: Query OK, 0 rows affected (0.001 sec)
-- PHP returns: 1 => Should be 0, one row selected, but no row affected
```

**Comment by Altahrim** [comment]
Finally, it seems `INSERT IGNORE` is the only way to achieve this with our current setup of MySQL.

From [MySQL documentation](https://dev.mysql.com/doc/refman/8.0/en/insert-on-duplicate.html)
> With ON DUPLICATE KEY UPDATE, the affected-rows value per row is 1 if the row is inserted as a new row, 2 if an existing row is updated, and 0 if an existing row is set to its current values. If you specify the CLIENT_FOUND_ROWS flag to the [mysql_real_connect()](https://dev.mysql.com/doc/c-api/8.0/en/mysql-real-connect.html) C API function when connecting to [mysqld](https://dev.mysql.com/doc/refman/8.0/en/mysqld.html), the affected-rows value is 1 (not 0) if an existing row is set to its current values.

The other way is to update all fields instead of ignore them but I failed to achieve this with Oracle. It's sad because it could make sense when we launch an update after the `insertIgnoreConflict`

**Comment by Altahrim** [comment]
@nickvergessen Do you think it is a correct solution given the risks of ignoring other errors?

**Review by come-nc — COMMENTED** [review]
A unit test would be nice

**Review by yemkareems — APPROVED** [review]

**Pull request merged**